sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -56,6 +56,66 @@ public class BlobBaseClient
         /// </summary>
         public virtual CustomerProvidedKey? CustomerProvidedKey => _customerProvidedKey;
 
+        /// <summary>
+        /// The Storage account name corresponding to the blob client.
+        /// </summary>
+        private string _accountName;
+
+        /// <summary>
+        /// Gets the Storage account name corresponding to the blob client.
+        /// </summary>
+        public string AccountName
+        {
+            get
+            {
+                if (_accountName == null)
+                {
+                    _accountName = new BlobUriBuilder(Uri).AccountName;
```

Comment:
If you're going to parse the Uri, you should get all the values out at once.  I'd add a `private void GetUriComponents()` method or similar that parses and sets _accountName, _containerName, and _name in one fell swoop.

Reply:
Ditto for the other clients.